bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs

```diff
@@ -77,92 +90,178 @@ impl<
+		}
 
-			if message.value > 0 {
-				// Asset for remaining ether
-				let remaining_ether_asset: Asset = (
-					ether_location.clone(),
-					message.value
-				).into();
-				assets.push(AssetTransfer::ReserveDeposit(remaining_ether_asset));
+		for asset in &message.assets {
+			match asset {
+				EthereumAsset::NativeTokenERC20 { token_id, value } => {
+					let token_location: Location = Location::new(
+						2,
+						[
+							GlobalConsensus(EthereumNetwork::get()),
+							AccountKey20 { network: None, key: (*token_id).into() },
+						],
+					);
+					let asset: Asset = (token_location, *value).into();
+					assets.push(AssetTransfer::ReserveDeposit(asset));
+				},
+				EthereumAsset::ForeignTokenERC20 { token_id, value } => {
+					let asset_loc = ConvertAssetId::convert(&token_id)
+						.ok_or(ConvertMessageError::InvalidAsset)?;
+					let mut reanchored_asset_loc = asset_loc.clone();
+					reanchored_asset_loc
+						.reanchor(&GlobalAssetHubLocation::get(), &EthereumUniversalLocation::get())
+						.map_err(|_| ConvertMessageError::CannotReanchor)?;
+					let asset: Asset = (reanchored_asset_loc, *value).into();
+					assets.push(AssetTransfer::ReserveWithdraw(asset));
+				},
 			}
+		}
 
-			for asset in &message.assets {
-				match asset {
-					EthereumAsset::NativeTokenERC20 { token_id, value } => {
-						let token_location: Location = Location::new(
-							2,
-							[
-								GlobalConsensus(EthereumNetwork::get()),
-								AccountKey20 { network: None, key: (*token_id).into() },
-							],
-						);
-						let asset: Asset = (token_location, *value).into();
-						assets.push(AssetTransfer::ReserveDeposit(asset));
-					},
-					EthereumAsset::ForeignTokenERC20 { token_id, value } => {
-						let asset_loc = ConvertAssetId::convert(&token_id)
-							.ok_or(ConvertMessageError::InvalidAsset)?;
-						let mut reanchored_asset_loc = asset_loc.clone();
-						reanchored_asset_loc
-							.reanchor(&GlobalAssetHubLocation::get(), &EthereumUniversalLocation::get())
-							.map_err(|_| ConvertMessageError::CannotReanchor)?;
-						let asset: Asset = (reanchored_asset_loc, *value).into();
-						assets.push(AssetTransfer::ReserveWithdraw(asset));
-					},
-				}
-			}
+		let topic = blake2_256(&("SnowbridgeInboundQueueV2", message.nonce).encode());
 
-			let topic = blake2_256(&("SnowbridgeInboundQueueV2", message.nonce).encode());
+		let prepared_message = PreparedMessage {
+			origin: message.origin.clone(),
+			claimer,
+			assets,
+			remote_xcm,
+			execution_fee: execution_fee_asset,
+			topic,
+		};
 
-			let prepared_message = PreparedMessage {
-				origin: message.origin.clone(),
-				claimer,
-				assets,
-				remote_xcm,
-				execution_fee: execution_fee_asset,
-				topic
-			};
+		Ok(prepared_message)
+	}
 
-			Ok(prepared_message)
+	/// Construct the `Xcm<()>` needed to create a new asset on Polkadot.
+	/// Returns an error if the source network is not Ethereum and the target network is not
+	/// Polkadot.
```

Comment:
Wording could be more descriptive
```suggestion
	/// Construct the remote XCM needed to create a new asset in the `ForeignAssets` pallet
	/// on AssetHub (Polkadot or Kusama).
```